### PR TITLE
feat: new aegir

### PR DESCRIPTION
**Pull request by hugomrdias**

### CODE REVIEW

File: test/utils/daemon-factory.js [hunk 1]
```diff
@@ -2,22 +2,20 @@
...
-  ipfsModule = require('ipfs')
-}
 
-if (process.env.IPFS_JS_HTTP_MODULE) {
+let ipfsHttpModule
+let ipfsModule
+try {
   ipfsHttpModule = require(process.env.IPFS_JS_HTTP_MODULE)
-} else {
+} catch (err) {
```
Comment (achingbrain) [inline test/utils/daemon-factory.js:10]: We should be able to use [optional catch binding](https://2ality.com/2017/08/optional-catch-binding.html) now?

```suggestion
} catch {
```

File: test/utils/daemon-factory.js [hunk 2]
```diff
@@ -2,22 +2,20 @@
...
   ipfsHttpModule = require('ipfs-http-client')
 }
 
+try {
+  ipfsModule = require(process.env.IPFS_JS_MODULE)
+} catch (err) {
```
Comment (achingbrain) [inline test/utils/daemon-factory.js:16]: Same as the other try/catch

```suggestion
} catch {
```

File: package.json [hunk 3]
```diff
@@ -63,6 +64,12 @@
     "rimraf": "^3.0.0",
     "uint8arrays": "^2.0.5"
   },
+  "eslintConfig": {
+    "extends": "ipfs",
+    "ignorePatterns": [
+      "!.aegir.js"
```
Comment (achingbrain) [inline package.json:70]: Why are we not linting this file?

**Review by achingbrain — CHANGES_REQUESTED** [review]
LGTM, a few small nits

File: package.json (hunk 3 above)
Comment (hugomrdias) [reply package.json:70]: this actually tell the linter to **lint** the file, double negative lol

Comment (hugomrdias) [reply package.json:70]: eslint ignores dot files by default we need to tell the linter to not ignore .aegir.js inside the ignore config 🤷🏻‍♂️

**Pull request merged**